kernel/mem_domain.c

```diff
@@ -155,6 +155,8 @@ void k_mem_domain_add_partition(struct k_mem_domain *domain,
 		}
 	}
 
+	__ASSERT((p_idx < arch_mem_domain_max_partitions_get()),
+		"no free partition slots available");
```

Comment:
This should not be necessary because of `max_partitions = arch_mem_domain_max_partitions_get()` at line 274.

Reply:
ah yes.... good point

Reply:
wait, I know during my debugging that I saw this value be something else because the underlying drivers for the NXP MPU had not been run yet to set the static_regions_num that is subtracted from the number of available regions. So the initial value would be 9 then after running it would be 4. 

But now I've just pulled, again, from the tip of master and the case is no longer failing when it was this morning and last night.

Do we have a race condition for when init_mem_domain_module() is called and when the MPU driver may get run to set the static fields?

Reply:
hm... `init_mem_domain_module()` is called via `SYS_INIT()` at `PRE_KERNEL_1`. This should be after the call to `arch_kernel_init()` in `arch/arm/include/aarch32/kernel_arch_func.h` where it calls `z_arm_mpu_init()`.

Reply:
`static_regions_num` is updated via `mpu_configure_static_mpu_regions()`. Maybe this is why?

Reply:
Did some bisecting and this PR #34086 moved some things around that may have fixed this. I'm trying to understand what I was using earlier this week when I originally bisected the problem because the tip should have had this PR in it.

Reply:
@dleach02 I think we don't need this, right?

Reply:
No we don't. Given the PR I referenced moved the initialization of the mpu_configure_static_mpu_regions() to occur before the PRE_KERNEL_1 initialization of the kernel/mem_domain.c:init_mem_domain_module() the variable in question is now initialized before being referenced.